core/metatable.d.ts

```diff
@@ -1,6 +1,10 @@
 // Based on https://www.lua.org/manual/5.3/manual.html#2.4
 
-interface LuaMetatable<T> {
+/**
+ * @typeparam T The prototype object and the metatable.
+ * @typeparam M This metatable.
+ */
```

Comment:
`prototype object` may be confusing and I don't really get what `and the metatable` means.
From `This metatable` I'd generally expect it to have `LuaMetatable`, not just one member of it.

But generally, I'm not really sure if comments are necessary there, I think with a better naming of `M` usage should be obvious.